commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java

```diff
@@ -0,0 +1,249 @@
+
+    @TempDir
+    Path projectDir;
+
+    @BeforeEach
+    void setUp() {
+        appModel = new ApplicationModelBuilder()
+                .setAppArtifact(ResolvedDependencyBuilder.newInstance()
+                        .setWorkspaceModule(WorkspaceModule.builder()
+                                .setModuleId(WorkspaceModuleId.of("org.acme", "app", "1.0"))
+                                .setModuleDir(projectDir)
+                                .build()))
+                .build();
+    }
+
+    private Config buildConfig() {
+        return ConfigProviderResolver.instance()
+                .getBuilder()
+                .withSources(new PropertiesConfigSource(configProperties, "test"))
+                .build();
+    }
+
+    @Test
+    void init_legacyFolder_setInputDirectory() throws IOException {
+        Path frontendFolder = projectDir.resolve("frontend");
+        Files.createDirectories(frontendFolder);
+
+        provider.init(appModel, Map.of());
+
+        assertThat(provider.getInputDirectory()).isEqualTo(frontendFolder);
+    }
+
+    @Test
+    void init_defaultFrontendFolder_setInputDirectory() throws IOException {
+        Path frontendFolder = projectDir.resolve(Path.of("src", "main", "frontend"));
+        Files.createDirectories(frontendFolder);
+
+        provider.init(appModel, Map.of());
+
+        assertThat(provider.getInputDirectory()).isEqualTo(frontendFolder);
+    }
+
+    @Test
+    void init_legacyAndDefaultFrontendFolders_setDefaultAsInputDirectory() throws IOException {
+        Path legacyFrontendFolder = projectDir.resolve("frontend");
+        Files.createDirectories(legacyFrontendFolder);
+        Path frontendFolder = projectDir.resolve(Path.of("src", "main", "frontend"));
+        Files.createDirectories(frontendFolder);
+
+        provider.init(appModel, Map.of());
+
+        assertThat(provider.getInputDirectory()).isEqualTo(frontendFolder);
+    }
+
+    @Test
+    @Disabled("Not sure about this test")
+    void init_customFrontendFolder_setInputDirectory() throws IOException {
+        String originalValue = System.getProperty(FrontendUtils.PARAM_FRONTEND_DIR);
+        String customFrontend = "custom/frontend";
+
+        Path frontendFolder = projectDir.resolve(Path.of("src", "main", "frontend"));
+        Files.createDirectories(frontendFolder);
+        Path customFrontendFolder = projectDir.resolve(Path.of("custom", "frontend"));
+        Files.createDirectories(customFrontendFolder);
+
+        System.setProperty(FrontendUtils.PARAM_FRONTEND_DIR, customFrontend);
+        try {
+            provider.init(appModel, Map.of());
+
+            assertThat(provider.getInputDirectory()).isEqualTo(customFrontendFolder);
+        } finally {
+            if (originalValue != null) {
+                System.setProperty(FrontendUtils.PARAM_FRONTEND_DIR, originalValue);
+            } else {
+                System.clearProperty(FrontendUtils.PARAM_FRONTEND_DIR);
+            }
+        }
+    }
```

Comment:
Whats not clear here, remove it or keep it?

Reply:
Since the code generation runs also separately from the application (own goal) I am not sure we should handle the case of system property setting frontend folder. It looks like a corner case, maybe just fix if someone complains

Reply:
I think thats fine. As we are im a quarkus domain, configuration is preferred to use other mechanisms. I am always a friend of supporting corner cases, but if the effort is to high and the propability to low, just keep it as it is.

Reply:
I propose to keep the disabled test for a while, and if we do not get feedback, remove it in a future release